app/src/main/java/com/example/internetshop/presentation/activity/MainActivity.kt

```diff
@@ -109,14 +101,23 @@ class MainActivity : AppCompatActivity(), ContainerHolder {
                 R.id.cartFragment
             )
         ).build()
+        binding?.let {
+            it.bottomNavBar.setupWithNavController(navController!!)
+            it.toolbar.setNavigationIcon(R.drawable.ic_back_arrow)
+            it.collapsingLayout.setupWithNavController(
+                it.toolbar,
+                navController!!,
+                appBarConfig!!
+            )
+        }
+
     }
 
     private fun subscribeToBottomNavVisibility() {
-        bottomNavViewModel.bottomNavVisibility.observe(this, {
-            if(it) {
+        bottomNavViewModel.visibility.observe(this, {
+            if (it) {
```

Comment:
In kotlin `if` is expression. This means that in kotlin if can return a result. Refactor this to 
`val visibility = if(...) View.VISIBLE else VIew.GONE
yourView.visibility = visibility`

Reply:
fixed